python-sdk/nuscenes/eval/lidarseg/render.py

```diff
@@ -0,0 +1,327 @@
+                                       sample_token: str,
+                                       lidarseg_preds_folder: str = None,
+                                       axes_limit: float = 40,
+                                       dot_size: int = 5,
+                                       out_path: str = None) -> None:
+    """
+    Visualize semantic difference of lidar segmentation results in bird's eye view.
+    :param nusc: A NuScenes object.
+    :param sample_token: Unique identifier.
+    :param lidarseg_preds_folder: A path to the folder which contains the user's lidar segmentation predictions for
+                                  the scene. The naming convention of each .bin file in the folder should be
+                                  named in this format: <lidar_sample_data_token>_lidarseg.bin.
+    :param axes_limit: Axes limit for plot (measured in meters).
+    :param dot_size: Scatter plot dot size.
+    :param out_path: Path to save visualization to (e.g. /save/to/here/bev_diff.png).
+    """
+    mapper = LidarsegClassMapper(nusc)
+
+    sample = nusc.get('sample', sample_token)
+
+    # Get the sample data token of the point cloud.
+    sd_token = sample['data']['LIDAR_TOP']
+    pointsensor = nusc.get('sample_data', sd_token)
+    pcl_path = os.path.join(nusc.dataroot, pointsensor['filename'])
+
+    # Load the ground truth labels for the point cloud.
+    gt_path = os.path.join(nusc.dataroot, nusc.get('lidarseg', sd_token)['filename'])
+    gt = LidarSegPointCloud(pcl_path, gt_path)
+    gt.labels = mapper.convert_label(gt.labels)  # Map the labels as necessary.
+
+    # Load the predictions for the point cloud.
+    preds_path = os.path.join(lidarseg_preds_folder, sd_token + '_lidarseg.bin')
+    preds = LidarSegPointCloud(pcl_path, preds_path)
+
+    # Do not compare points which are ignored.
+    ignored_points_idxs = np.where(gt.labels != mapper.ignore_class['index'])[0]
+    gt.labels = gt.labels[ignored_points_idxs]
+    gt.points = gt.points[ignored_points_idxs]
+    preds.labels = preds.labels[ignored_points_idxs]
+    preds.points = preds.points[ignored_points_idxs]
+
+    # Init axes.
+    fig, axes = plt.subplots(1, 3, figsize=(10 * 3, 10), sharex='all', sharey='all')
+
+    # Render ground truth and predictions.
+    gt.render(mapper.coarse_colormap, mapper.coarse_name_2_coarse_idx_mapping, ax=axes[0])
+    preds.render(mapper.coarse_colormap, mapper.coarse_name_2_coarse_idx_mapping, ax=axes[1])
+
+    # Render errors.
+    id2color_for_diff_bev = {0: (191, 41, 0, 255),  # red: wrong label
+                             1: (50, 168, 82, 255)}  # green: correct label
+    colors_for_diff_bev = colormap_to_colors(id2color_for_diff_bev, {0: 0, 1: 1})
+    mask = np.array(gt.labels == preds.labels).astype(int)  # Convert array from bool to int.
+    axes[2].scatter(gt.points[:, 0], gt.points[:, 1], c=colors_for_diff_bev[mask], s=dot_size)
+    axes[2].set_title('Errors (Correct: Green, Mislabeled: Red)')
```

Comment:
Reduce the white space between plots and around all of them.

Reply:
Some commands that might help:
```
        plt.savefig(out_path, bbox_inches='tight', pad_inches=0, dpi=200)
        plt.tight_layout()
```

Reply:
Done